### PR TITLE
[wxWidgets] update to 3.1.5

### LINKED ISSUE #17313: [wxWidgets] update to 3.1.5
Library name: `wxwidgets`

New version number: `3.1.5`

Other information that may be useful (release notes, etc...)
https://github.com/wxWidgets/wxWidgets/releases/tag/v3.1.5

https://www.wxwidgets.org/news/2021/04/wxwidgets-3.1.5-released/

### PR DESCRIPTION
**Pull request by electron271**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17313

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  UWP is not supported

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `I think so..`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 No

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by electron271** [comment]
how exactly to update the patch?

**Comment by SamuelMarks** [comment]
@electron271 You might also want to look at #17072 and #17111

**Comment by JonLiu1993** [comment]
@electron271 ,usually update the port, there are the following steps:
1. Convert the control file to the vcpkg.json file, and modify the version inside to the target version. If the version is a special combination method, you need to use version, version-string, version-semver or version-date respectively. For details, please refer to [here](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#versioning). If there is port-version before update, it needs to be set to 0 or deleted.
2. Modify the ref and hash value in portfile.cmake.
3. If you need to update the patch in the port, you need to investigate the port history update record and combine the target version in the soure to remake the patch.

- If the previous patch has been successfully added to the source of the target version, you can delete the patch.
- If the latest version of the source does not contain the modifications in the patch and the patch cannot be used normally, you need to remake the patch.
- If the latest version of the source does not contain the patch modification content, the patch can be used directly, just keep the patch intact and use it directly.

The above is the most basic way to update ports. Of course, this is the only way to update all ports. We have more small details and gradual update strategies that need to be paid attention to at any time. Hope these can help you, thanks for your contribution

File: ports/wxwidgets/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "wxwidgets",
-  "version-semver": "3.1.4",
-  "port-version": 7,
+  "version-semver": "3.1.5",
+  "port-version": 8,
```
Comment (JonLiu1993) [inline ports/wxwidgets/vcpkg.json:4]: @electron271 ,please delete port-version or set port-version to 0

File: versions/w-/wxwidgets.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "b284aa561d3ec113a892e6fb37e4b4333bf60517",
+      "version-semver": "3.1.5",
+      "port-version": 8
```
Comment (JonLiu1993) [inline versions/w-/wxwidgets.json:6]: please delete port-version or set port-version to 0

**Comment by JonLiu1993** [comment]
@electron271 ,I found that the current problem of CI is due to the incorrect use of patch. You can try to remake the patch.
````
Applying patch fix-stl-build-vs2019-16.6.patch
CMake Error at scripts/cmake/z_vcpkg_apply_patches.cmake:57 (message):
  Applying patch failed: Checking patch include/wx/hashmap.h...

````

**Comment by electron271** [comment]
how would I remake the patch?

**Comment by JonLiu1993** [comment]
> how would I remake the patch?

@electron271 ,You can follow the steps below

Direct   to the libs path src: such as :C:\fixDate\vcpkg\buildtrees\date\src\date-2.4
--
Git init .
Git add .
Git commit -m  "test amp"
Modify source
git   diff  | out-file -enc ascii fix-.patch

**Pull request closed without merging**